### PR TITLE
CloudSQLite: If user already has write lock and calls acquireWriteLock again, then don't release it.

### PR DESCRIPTION
**Pull request by nick4598**

### CODE REVIEW

File: core/backend/src/CloudSqlite.ts [hunk 1]
```diff
@@ -623,13 +623,16 @@ export namespace CloudSqlite {
     await acquireWriteLock(args);
```
Comment (kabentley) [inline core/backend/src/CloudSqlite.ts:623]: I think it's easier to just do:

```ts
    if (args.container.hasWriteLock)
      return operation();
```

Comment (nick4598) [reply core/backend/src/CloudSqlite.ts:623]: if (args.container.hasWriteLock && heldBySameUser) ?

Comment (nick4598) [reply core/backend/src/CloudSqlite.ts:623]: Also, if I'm not mistaken the current state of the code does not allow you to refresh the write lock if you're the same user in the case of a long-running operation? CloudSqlite.acquireWriteLock returns if its the same user, but the native version of acquireWriteLock has

 "Calling `acquireWriteLock` with the lock already held resets the lock duration from the current time,
     * so long running processes should call this method periodically to ensure their lock doesn't expire (they should also make sure their accessToken is refreshed
     * before it expires.)"

Comment (kabentley) [reply core/backend/src/CloudSqlite.ts:623]: there's only one write lock, so if it's already held we should do what the comment says. But the whole "same user" thing is only relevant for RPC calls, which can't be recursive or long-running.

**Review by kabentley — APPROVED** [review]

File: core/backend/src/CloudSqlite.ts [hunk 2]
```diff
@@ -623,6 +623,8 @@ export namespace CloudSqlite {
...
     try {
       const containerInternal = args.container as CloudContainerInternal;
+      if (containerInternal.writeLockHeldBy === args.user) // If the user already had the write lock, then don't release it.
```
Comment (kabentley) [inline core/backend/src/CloudSqlite.ts:626]: oops sorry. You need to save this BEFORE calling acquireWriteLock,

**Review by kabentley — CHANGES_REQUESTED** [review]

File: core/backend/src/CloudSqlite.ts (hunk 2 above)
Comment (kabentley) [reply core/backend/src/CloudSqlite.ts:626]: nevermind, it's fine the way you have it (this is all tricky).

**Review by kabentley — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]

**Pull request merged**